apps/extension/src/routes/page/onboarding/generate.tsx

```diff
@@ -14,20 +14,74 @@ import { generateSelector } from '../../../state/seed-phrase/generate';
 import { usePageNav } from '../../../utils/navigate';
 import { PagePath } from '../paths';
 import { WordLengthToogles } from '../../../shared/containers/word-length-toogles';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { fetchBlockHeight } from '../../../hooks/full-sync-height';
+import { localExtStorage } from '../../../storage/local';
+import { sample } from 'lodash';
+
+const fetchBlockHeightWithFallback = async (endpoints: string[]): Promise<number> => {
+  if (endpoints.length === 0) {
+    throw new Error('All RPC endpoints failed to fetch the block height.');
+  }
+
+  // Randomly select an rpc endpoint from the chain registry
+  const randomEndpoint = sample(endpoints);
+
+  try {
+    const walletCreationBlockHeight = await fetchBlockHeight(randomEndpoint!);
+    if (walletCreationBlockHeight !== undefined) {
+      return walletCreationBlockHeight;
```

Comment:
issue: This is related to my other comment. If there is an error, it will go to the catch clause. Don't think we need to return undefined from the function. This will also prevent the need to have the double `const remainingEndpoints...` lines.

Reply:
modified logic to 

```ts
  const walletCreationBlockHeight = await fetchBlockHeight(randomGrpcEndpoint);

  if (walletCreationBlockHeight !== undefined) {
    return walletCreationBlockHeight;
  } else {
    // Remove the current endpoint from the list and retry with remaining endpoints
    const remainingEndpoints = endpoints.filter(endpoint => endpoint !== randomGrpcEndpoint);
    return fetchBlockHeightWithFallback(remainingEndpoints);
  }
```

It's essentially saying that if the tendermint client returns `undefined` or an error occurs, we want it will recursively attempt another endpoint, otherwise return the block height.

Reply:
modified to incorporate more explicit error handling, rather than clobbering errors as undefined and recursing.